interfaces/IApplication.h

```diff
@@ -23,17 +24,79 @@
 namespace WPEFramework {
 namespace Exchange {
 
-    // This interface gives direct access to an Application instance
+    // @json
     struct IApplication : virtual public Core::IUnknown {
 
         enum { ID = ID_APPLICATION };
 
-
         virtual ~IApplication() {}
 
-        virtual void Reset() = 0;
-        virtual void DeepLink(const string& deepLink) = 0;
+        enum resettype : uint8_t {
+            FACTORY,
+            CACHE,
+            CREDENTIALS,
+            RECORDINGS
+        };
+
+        enum launchpointtype : uint8_t {
+            UNDEFINED,
+            DIAL,
+            DEDICATED_BUTTON,
+            DEDICATED_ICON,
+            APPLICATION_LIST,
+            INTEGRATED_TILE,
+            SEARCH_RESULT,
+            SEARCH_CONTINUATION,
+            VOICE_CONTROL,
+            VOICE_SEARCH_RESULT,
+            VISUAL_GESTURE,
+            TOUCH_GESTURE,
+            EPG_GRID,
+            CHANNEL_NUMBER,
+            CHANNEL_ZAP,
+            CHANNEL_BAR,
+            WEB_BROWSER,
+            POWER_ON,
+            POWER_ON_FROM_DEDICATED_BUTTON
+            SUSPENDED_POWER_ON,
+            RESTART,
+            SUSPENDED_RESTART,
+            RESUMED_FROM_SCREENSERVER,
+            RESUMED_FROM_STANDBY,
+            BANNER_AD,
+            TITLE_RECOMENDATION,
+            APPLICATION_PROMOTION
+        };
+
+        // @brief Resets application data
+        // @param type Type of reset to perform
+        virtual uint32_t Reset(const resettype type) = 0;
+
+        // @property
+        // @brief Application-specific identification value
+        // @param id ID string (e.g. ABCDEFG=00000005)
+        virtual uint32_t ID(string& id /* @out */) const = 0;
```

Comment:
Showing error, since both enum and property use same name 'ID"

Reply:
Renamed to Identifier